File: KahootClient/src/SocialUser.ts
export class SocialUser{
  public id: number | undefined;

  constructor(public userName: string | undefined,
              public name: string | undefined,
              public email: string | undefined,
              public role: string | undefined,
              public provider: string | undefined) {  }

  public get Id() {
    return this.id;
  }

  public get Username() {
    return this.userName;
  }

  public get Name() {
    return this.name;
  }

  public get Email() {
    return this.email;
  }

  public get Role() {
    return this.role;
  }
  public get Provider() {
    return this.provider;
  }
}

File: KahootWebApi/Models/SocialUser.cs
using System.ComponentModel.DataAnnotations;

namespace KahootWebApi.Models
{
    public class SocialUser
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string? Username { get; set; }

        public string? Name { get; set; }

        [Required]
        public string? Email { get; set; }

        [Required]
        public string? Role { get; set; }

        public string? Provider { get; set; }

        [Required]
        public int UserId { get; set; }

        public User? User { get; set; }
    }
}
